presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -480,6 +481,12 @@ public StreamProperties visitApply(ApplyNode node, List<StreamProperties> inputP
         }
 
         @Override
+        public StreamProperties visitLateralJoin(LateralJoinNode node, List<StreamProperties> inputProperties)
+        {
+            return inputProperties.get(0);
```

Comment:
ditto. See `visitJoin` in this class and make sure derived properties are correct